arangod/RestServer/Metric/Batch.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2021 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Valery Mironov
+////////////////////////////////////////////////////////////////////////////////
+
+#include <RestServer/Metric/Metric.h>
+
+#include <mutex>
+
+namespace arangodb::metric {
+
+template <typename T>
+class Batch final : public Metric {
+ public:
+  Batch(T&& metric, std::string_view name, std::string_view help, std::string_view labels)
+      : Metric{name, help, labels}, _metric{std::move(metric)} {}
+
+  [[nodiscard]] std::string_view type() const noexcept final {
+    return "untyped";
+  }
+  void toPrometheus(std::string& result, std::string_view globals, std::string_view) const final {
```

Comment:
The `final` is not necessary here, as the class is already marked `final`. I suggest we add `override` here because otherwise compilers may warn about it being missing.
```suggestion
  void toPrometheus(std::string& result, std::string_view globals, std::string_view) const override {
```

Reply:
What compiler says that `override` is needed when there is `final`? Can you give an example from [godbolt](https://godbolt.org/)?

Reply:
Motivation to use `final` instead of `override` then possible (from a semantic point of view, they mean the same thing, except for the case of the most basic `virtual` function, you can write `final` there, but you cannot `override`, but if you never write `final` together with `virtual` (this is pointless), this problem doesn't exist):
More information for the compiler, you are right that in this case, it is not so, but if at some time our `class` becomes not `final`, the method that was `final` in its meaning will cease to be so

Reply:
Ok, it seems you are right. There doesn't need to be an extra `override` specifier for methods that are marked `final`.
So it should work fine as it is in the PR right now. I prefer the other variant (declaring just the class `final` but not all methods, and instead mark overriden methods as `override`), but that is just a matter of taste.
So let's leave it as it is.